packages/wrangler/src/pages/deploy.tsx

```diff
@@ -337,9 +337,42 @@ export const Handler = async (args: PagesDeployArgs) => {
 		project_name: projectName,
 	});
 
-	logger.log(
-		`✨ Deployment complete! Take a peek over at ${deploymentResponse.url}`
-	);
+	let latestDeploymentStage: DeploymentStage | undefined;
+
+	// can this ever become an infinite loop?
+	while (
+		latestDeploymentStage?.name !== "deploy" &&
+		latestDeploymentStage?.status !== "success" &&
+		latestDeploymentStage?.status !== "failure"
+	) {
+		try {
+			const deployment = await fetchResult<Deployment>(
+				`/accounts/${accountId}/pages/projects/${projectName}/deployments/${deploymentResponse.id}`
+			);
+			latestDeploymentStage = deployment.latest_stage;
+		} catch (err) {
+			logger.debug(
+				`Attempt to get deployment status for deployment with id "${deploymentResponse.id}" failed: ${err}`
+			);
+		}
+	}
+
+	if (latestDeploymentStage.status === "success") {
```

Comment:
@CarmenPopoviciu if we fail to get the deployment details repeatedly for some reason, and we set a max attempts cap, we'll exit that `while` loop in a weird state, with an unknown deployment status. We should resolve that status one way or the other (`success` | `failure`) so we can show the appropriate user msg

Reply:
@WalshyDev or @tanushree-sharma would appreciate some input here on what the desired behaviour is 🙏

Reply:
```
Deployment complete. However, we couldn't ascertain the final status of your deployment.

Visit your deployment: ${deploymentResponse.url}
Check logs on the Cloudflare dashboard here: https://dash.cloudflare.com/xyz/projects/xyz/${deploymentResponse.id}
```

Reply:
like the wording very much. Thank you thank you! Will add!

Reply:
✅